hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/SelectProcessorSupplier.java

```diff
@@ -0,0 +1,183 @@
+import static com.mongodb.client.model.Projections.include;
+import static java.util.Arrays.asList;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * ProcessorSupplier that creates {@linkplain com.hazelcast.jet.mongodb.impl.ReadMongoP} processors on each instance.
+ */
+public class SelectProcessorSupplier implements ProcessorSupplier {
+
+    private final String connectionString;
+    private final String databaseName;
+    private final String collectionName;
+    private final boolean stream;
+    private final FunctionEx<ExpressionEvalContext, EventTimePolicy<JetSqlRow>> eventTimePolicyProvider;
+    private final String predicate;
+    private final List<String> projection;
+    private final Long startAt;
+    private transient ExpressionEvalContext evalContext;
+    private final QueryDataType[] types;
+
+    SelectProcessorSupplier(MongoTable table, String predicate, List<String> projection, Long startAt, boolean stream,
+                            FunctionEx<ExpressionEvalContext, EventTimePolicy<JetSqlRow>> eventTimePolicyProvider) {
+        this.predicate = predicate;
+        this.projection = projection;
+        this.connectionString = table.connectionString;
+        this.databaseName = table.databaseName;
+        this.collectionName = table.collectionName;
+        this.startAt = startAt;
+        this.stream = stream;
+        this.eventTimePolicyProvider = eventTimePolicyProvider;
+
+        checkArgument(projection != null && !projection.isEmpty(), "projection cannot be empty");
+        this.types = table.resolveColumnTypes(projection);
+    }
+
+
+    SelectProcessorSupplier(MongoTable table, String predicate, List<String> projection, Long startAt,
+                            FunctionEx<ExpressionEvalContext, EventTimePolicy<JetSqlRow>> eventTimePolicyProvider) {
+        this(table, predicate, projection, startAt, true, eventTimePolicyProvider);
+    }
+
+    SelectProcessorSupplier(MongoTable table, String predicate, List<String> projection) {
+        this(table, predicate, projection, null, false, null);
+    }
+
+    @Override
+    public void init(@Nonnull Context context) {
+        evalContext = ExpressionEvalContext.from(context);
+    }
+
+    @Nonnull
+    @Override
+    public Collection<? extends Processor> get(int count) {
+        ArrayList<Bson> aggregates = new ArrayList<>();
+
+        if (this.predicate != null) {
+            Document filterDoc = Document.parse(this.predicate);
+            Bson filterWithParams = ParameterReplacer.replacePlaceholders(filterDoc, evalContext);
+            aggregates.add(match(filterWithParams.toBsonDocument()));
+        }
+        Bson proj = include(this.projection);
+        if (!projection.contains("_id") && !stream) {
+            aggregates.add(project(fields(excludeId(), proj)));
+        } else {
+            aggregates.add(project(proj));
+        }
+
+        Processor[] processors = new Processor[count];
```

Comment:
Maybe add all processors directly to ArrayList instead of array and then converting it back to collection